Improved filtering for ballistic simulations

almaeder opened the pull request:
This add an extra condition to capture slow decaying modes in the obc filtering.
This helps to reduce the recursion error in the obc in the ballistic case.

almaeder committed:
Improved filtering for ballistic

maurodossena commented on src/qttools/obc/spectral.py:
I would add the following condition also (to match the condition for propagating modes):

```
mask_injected &= self.min_propagation < abs(dEk_dk.real) / (
            abs(dEk_dk.imag) + eta
        )
```

almaeder committed:
Extra injection condition

This add an extra condition for the injected modes
to match the ones for reflected modes.

maurodossena reviewed (approved):
Looks good!
